src/mmds/src/lib.rs

```diff
@@ -11,18 +11,36 @@ use serde_json::{Map, Value};
 use std::sync::{Arc, Mutex};
 
 use crate::data_store::{Error as MmdsError, Mmds, MmdsVersion, OutputFormat};
+use crate::token::PATH_TO_TOKEN;
 
+use crate::token_headers::REJECTED_HEADER;
 use lazy_static::lazy_static;
-use micro_http::{Body, MediaType, Method, Request, Response, StatusCode, Version};
+use logger::warn;
+use micro_http::{
+    Body, HttpHeaderError, MediaType, Method, Request, RequestError, Response, StatusCode, Version,
+};
 use token_headers::TokenHeaders;
 
 pub const MAX_DATA_STORE_SIZE: usize = 51200;
+pub const NO_MMDS_ERROR: &str = "MMDS was not successfully created and \
+                                 it is not currently available for requests.";
+pub const NO_TOKEN_PROVIDED_ERROR: &str = "No MMDS token provided. Use `X-metadata-token` \
+                                           header to specify the session token.";
+pub const NO_TTL_PROVIDED_ERROR: &str = "Token time to live value not found. Use \
+                                         `X-metadata-token-ttl_seconds` header to specify \
+                                         the token's lifetime.";
 
 lazy_static! {
     // A static reference to a global Mmds instance. We currently use this for ease of access during
     // prototyping. We'll consider something like passing Arc<Mutex<Mmds>> references to the
     // appropriate threads in the future.
-    pub static ref MMDS: Arc<Mutex<Mmds>> = Arc::new(Mutex::new(Mmds::default()));
+    pub static ref MMDS: Option<Arc<Mutex<Mmds>>> = Mmds::new().map_or_else(
```

Comment:
I propose to remove the `Option` here. I was thinking that since we do support both MMDSv1 and MMDSv2 for now, customers that still prefer to use v1 should not be affected by MMDS default (v2) not being able to be initialized (for some reason or another). We could have MMDSv2 to be the default (as we currently do) and MMDSv1 the fallback mechanism in case MMDSv2 can't be initialized. We could log a warning if this happens. The user would be aware of the version configured because for MMDSv1 we do not allow `PUT` requests, so there would be no way to generate the token required for MMDSv2. WDYT?

Reply:
The `Option` is the result of [this comment](https://github.com/firecracker-microvm/firecracker/pull/2804#discussion_r765807994) and I'd keep it. The question is what do we do with it.

I am actually against a fallback route in this case. It's cumbersome to support both versions in a client. Users which start the MMDS expect to get MMDSv2 (either they explicitly set the version or the default version kicked in) and a fallback mechanism to MMDSv1 is bad CX in my opinion, as the user wants the v2 features. If they didn't care about the v2 features, they would use v1 from the start. It also complicates the whole setup by adding a "negotiation" phase.

If we decide to not start with MMDS by default, in my opinion it makes sense to crash the process early if we can't create it. If we don't change the MMDS behavior now (i.e. we start Firecracker with MMDS enabled by default) we can keep the current approach in this PR. It is highly unlikely that we won't be able to create the MMDS anyway, so this behavior would only affect the CX in the unlikely case of an error.